widget/embedded/src/components/SwapDetails/SwapDetails.tsx

```diff
@@ -0,0 +1,495 @@
+
+  const currentStep = swap ? getCurrentStep(swap) : ({} as PendingSwapStep);
+
+  const currentStepNetworkStatus = currentStep?.networkStatus;
+
+  useEffect(() => {
+    if (swap?.status === 'success' || swap?.status === 'failed') {
+      setShowCompletedModal(swap?.status);
+    }
+  }, [swap?.status]);
+
+  useEffect(() => {
+    if (swap && showSwitchNetwork) {
+      setModalState(PendingSwapNetworkStatus.WaitingForNetworkChange);
+    } else if (
+      currentStepNetworkStatus ===
+        PendingSwapNetworkStatus.WaitingForConnectingWallet ||
+      currentStepNetworkStatus === PendingSwapNetworkStatus.NetworkChanged
+    ) {
+      setModalState(currentStepNetworkStatus);
+    } else {
+      setModalState(null);
+    }
+  }, [currentStepNetworkStatus]);
+
+  if (!swap) {
+    return (
+      <SwapDetailsPlaceholder requestId={requestId || ''} loading={loading} />
+    );
+  }
+
+  const lastConvertedTokenInFailedSwap =
+    getLastConvertedTokenInFailedSwap(swap);
+
+  const currentStepBlockchain = currentStep
+    ? getCurrentBlockchainOfOrNull(swap, currentStep)
+    : null;
+  const currentStepWallet = currentStep
+    ? getRelatedWalletOrNull(swap, currentStep)
+    : null;
+
+  const percentageChange =
+    !swap.inputAmount || !swap.steps?.[swap.steps.length - 1]?.outputAmount
+      ? null
+      : getPercentageChange(
+          swap.inputAmount,
+          swap.steps[swap.steps.length - 1].outputAmount
+        );
+
+  const swapDate = getSwapDate(swap);
+
+  const shouldRetry = shouldRetrySwap(swap);
+  const isMobileWallet = (walletType: string): boolean =>
+    !!getWalletInfo(walletType)?.mobileWallet;
+
+  const showSwitchNetwork =
+    currentStepNetworkStatus ===
+      PendingSwapNetworkStatus.WaitingForNetworkChange &&
+    !!currentStepBlockchain &&
+    !!currentStepWallet?.walletType &&
+    (isMobileWallet(currentStepWallet.walletType) ||
+      canSwitchNetworkTo(currentStepWallet.walletType, currentStepBlockchain));
+
+  const switchNetwork = showSwitchNetwork
+    ? connect.bind(null, currentStepWallet.walletType, currentStepBlockchain)
+    : undefined;
+
+  const stepMessage = getSwapMessages(swap, currentStep);
+
+  const steps = getSteps(
+    swap,
+    switchNetwork,
+    currentStepNetworkStatus,
+    setModalState,
+    stepMessage
+  );
+  const numberOfSteps = steps.length;
+  const totalFee = swap.steps
+    .map((steps) => parseFloat(steps.feeInUsd || ''))
+    .reduce((totalFee, fee) => totalFee + fee);
+
+  const totalTime = swap.steps
+    .map((step) => step.estimatedTimeInSeconds || 0)
+    .reduce((totalTime, estimatedTime) => totalTime + estimatedTime);
+
+  const minutes = Math.floor(totalTime / SECONDS);
+
+  const firstStep = swap.steps[0];
+  const lastStep = swap.steps[numberOfSteps - 1];
+  const hasAtleastOneSuccessOrFailed = !!swap.steps.find(
+    (step) => step.status === 'failed' || step.status === 'success'
+  );
+
+  return (
+    <Layout
```

Comment:
I think you could refactor different parts of this page into different components to make code more readable. Based on time needed for the refactoring, we could decide.

Reply:
some sections have been refactored.

Reply:
some sections have been refactored.